### PR TITLE
split doSendMessage in sub methods

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: libtextsecure/outgoing_message.js [hunk 1]
```diff
@@ -293,309 +293,272 @@ OutgoingMessage.prototype = {
...
 
-        if (enableFallBackEncryption) {
-          sessionCipher = new libloki.crypto.FallBackSessionCipher(address);
-        } else {
-          sessionCipher = signalCipher;
-        }
-        const plaintext = this.getPlaintext(messageBuffer);
+    // Check if we need to attach the preKeys
+    const enableFallBackEncryption =
+      !keysFound || thisDeviceMessageType === 'friend-request';
```
Comment (Mikunj) [inline libtextsecure/outgoing_message.js:344]:
```suggestion
      !keysFound || thisDeviceMessageType === 'friend-request' || isMultiDeviceRequest;
```
i think you forgot this

File: libtextsecure/outgoing_message.js [hunk 2]
```diff
@@ -293,309 +293,272 @@ OutgoingMessage.prototype = {
...
+      const ciphertext = await secretSessionCipher.encrypt(
+        address,
+        senderCert,
+        plaintext,
+        sessionCipher
+      );
+      type = textsecure.protobuf.Envelope.Type.UNIDENTIFIED_SENDER;
+      content = window.Signal.Crypto.arrayBufferToBase64(ciphertext);
+    } else {
+      // TODO: probably remove this branch once
```
Comment (Mikunj) [inline libtextsecure/outgoing_message.js:454]: Now that we always use sealed sender, thoughts on removing all the other branches and simplifying this logic?

File: libtextsecure/outgoing_message.js (hunk 1 above)
Comment (Bilb) [reply libtextsecure/outgoing_message.js:344]: in fact, if you check when `isMultiDeviceRequest` is set to `true` in the old code, you will find that the `thisDeviceMessageType` is also set to `'friend-request'` at the samre time, so this is redundant

Comment (Mikunj) [reply libtextsecure/outgoing_message.js:344]: You are correct 👍

**Pull request merged**